File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
const input = fs.readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")


function solution([n, ...args]) {
  let digits = args.map(Number)
  let [dasom, ...others] = digits
  others.sort((a, b) => b - a)
  let max = others[0]
  if(max < dasom){
    console.log(0)
    return
  }
  let count = 0
  others = others.filter(x => x > dasom)
  while (max >= dasom) {
    dasom++
    count++
    others[0]--
    others.sort((a, b) => b - a)
    max = others[0]

  }
  console.log(count)
}

solution(input)

File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.kt
import java.io.BufferedReader
import java.io.File
import java.io.InputStreamReader
import java.util.*

fun main() {
    val os = System.getProperty("os.name").lowercase(Locale.getDefault())
    val args: List<String> = if (os.contains("linux")) {
        val br = BufferedReader(InputStreamReader(System.`in`))
        br.readLines()
    } else {
        val inputFile = File("src/main/kotlin/input.txt")
        inputFile.readLines()
    }

    val n = args[0].toInt()
    var dasom = args[1].toInt()
    val digits = args.slice(2 until args.size).map { it.toInt() }
    val pq = PriorityQueue<Int>(compareByDescending { it })
    pq.addAll(digits)
    var max = pq.poll()
    if(max == null || dasom > max){
        println(0)
        return
    }
    var count = 0
    while(dasom <= max){
        pq.offer(max - 1)
        dasom++
        count++
        max = pq.poll()
    }
    println(count)
}
